### PR TITLE
Ensure AnalyzerDriver completes during Compilation

### LINKED ISSUE #1086: Cancel AnalyzerDriver at the end of a compilation
This method deals with the treatment of `AnalyzerDriver` within `CommonCompiler.RunCore`.  

In the case that the compilation fails early, say in declaration or member binding, the compilation will not fully populate the event queue.  This means that `AnalyzerDriver` will not complete as it has several tasks waiting for the queue to complete.  The driver and all of the associated analyzers will all be stuck around waiting for more input that is never going to come.  So long as the Compilation remains in memory so will these components.  

The `RunCore` method should be ensuring here that the `AnalyzerDriver` completes no matter the success / failure of the underlying compilation.  This will help ensure prompt cleanup of resources.  

Note: Waiting for #1032 to complete before fixing this as it is heavily changing the same method.

### PR DESCRIPTION
**Pull request by jaredpar**
Today the AnalyzerDriver is only guaranteed to be completed after a
Compilation if the compilation made it to the Emit phase.  If the
compiler exited early, say to method body binding failures, the compiler
would not produce all of the compilation events and hence the analyzer
driver would be left forever waiting for them to complete.

closes #1086

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @VSadov , @AlekseyTs, @VladimirReshetnikov , @agocke

File: src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs [hunk 1]
```diff
@@ -328,159 +328,174 @@ private int RunCore(TextWriter consoleOutput, CancellationToken cancellationToke
...
-
-            // Print the diagnostics produced during the parsing stage and exit if there were any errors.
-            if (PrintErrors(compilation.GetParseDiagnostics(), consoleOutput))
-            {
-                return Failed;
-            }
-
-            if (PrintErrors(compilation.GetDeclarationDiagnostics(), consoleOutput))
+            CancellationTokenSource analyzerCts = null;
+            try
```
Comment (gafter) [inline src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs]: This looks good but it occurs to me... perhaps this method is getting too long?

**Comment by gafter** [comment]
:+1:

File: src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs (hunk 1 above)
Comment (jaredpar) [inline src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs]: I agree.  It's gotten a bit out of hand and needs to be broken up.  Will take a shot at it.

**Comment by VSadov** [comment]
LGTM

**Pull request merged**